cloud/blockstore/libs/storage/disk_registry/disk_registry_actor_switch_agent_to_read_only.cpp

```diff
@@ -20,16 +20,12 @@ void TDiskRegistryActor::HandleSwitchAgentDisksToReadOnly(
 {
     auto* msg = ev->Get();
 
-    auto it = AgentRegInfo.find(msg->AgentId);
-    if (it == AgentRegInfo.end()) {
-        ReportDiskRegistryAgentNotFound(
-            TStringBuilder()
-            << "HandleSwitchAgentDisksToReadOnly:AgentId: " << msg->AgentId);
-        return;
-    }
-
-    if (it->second.Connected) {
-        LOG_INFO(ctx, TBlockStoreComponents::DISK_REGISTRY,
+    if (auto it = AgentRegInfo.find(msg->AgentId);
```

Comment:
What if `SwitchAgentDisksToReadOnly` event was scheduled, and shortly after that the agent was removed from configuration? ExecuteTx<TSwitchAgentDisksToReadOnly>will be triggered on nonexistent agent?

Reply:
Yes, but it's not a problem: SwitchAgentDisksToReadOnly returns E_NOT_FOUND